File: CSES/2. Sorting and Searching/CSES-1619_Restaurant Customers.py
n = int(input())
times1, times2 = [], []
for _ in range(n):
    st, ed = map(int, input().split())
    times1.append(st)
    times2.append(ed)

times1.sort()
times2.sort()

i = j = 0
ans = cur = 0
while i < n and j < n:
    if i < n and times1[i] < times2[j]:
        cur += 1
        i += 1
    elif j < n and times1[i] > times2[j]:
        cur -= 1
        j += 1
    else:
        i += 1
        j += 1
    ans = max(ans, cur)

print(ans)

File: CSES/2. Sorting and Searching/CSES-1619_Restaurant Customers.cpp
#include <bits/stdc++.h>
using namespace std;
#define endl '\n'

int main() {
    ios::sync_with_stdio(false); cin.tie(nullptr); cout.tie(nullptr);
    int n, st, ed;
    cin >> n;
    map<int, int> d;
    for (int i = 0; i < n; i++) {
        cin >> st >> ed;
        d[st]++;
        d[ed]--;
    }
    int ans = 0, cur = 0;
    for (auto &[k, v]: d) {
        cur += v;
        ans = max(ans, cur);
    }
    cout << ans << endl;
    return 0;
}